server/s3_object_metadata.cc

```diff
@@ -1679,92 +1691,156 @@ bool S3ObjectMetadata::check_bucket_replication_policy(
+      if (destination_bucket_map.find(new_dest_bucket_name) !=
+          destination_bucket_map.end()) {
+        s3_log(S3_LOG_DEBUG, request_id, "same dest bucket \n");
+        rule_priority =
+            destination_bucket_map[new_dest_bucket_name]["Priority"];
+        if (new_rule_priority > rule_priority) {
+          destination_bucket_map[new_dest_bucket_name]
+                                ["DeleteMarkerReplicationStatus"] =
+                                    new_delete_marker_rep_status;
+          destination_bucket_map[new_dest_bucket_name]["Priority"] =
+              new_rule_priority;
+        }
+      } else {
+        s3_log(S3_LOG_DEBUG, request_id, "different dest bucket \n");
+        // Store destination bucket name in map as different destination bucket
+        // name present in different rules.
+        // Storing destination bucket names along with  deleteMarker status and
+        // priority in map.
+        destination_bucket_map[new_dest_bucket_name]
+                              ["DeleteMarkerReplicationStatus"] =
+                                  new_delete_marker_rep_status;
+        destination_bucket_map[new_dest_bucket_name]["Priority"] =
+            new_rule_priority;
+      }
+    } else {
+      s3_log(S3_LOG_DEBUG, request_id, "Filter does not matched.\n");
+      return false;
     }
-    if (!rule_object["Filter"].isNull()) {
-
-      if (!rule_object["Filter"]["And"]["Tag"].isNull() &&
-          rule_object["Filter"]["And"]["Prefix"].isNull()) {
-        // If tag,and nodes are present in filter and Prefix is not present.
-        Json::Value tag_array = rule_object["Filter"]["And"]["Tag"];
-        get_tags_from_replication_policy(rep_config_object_tags_map, tag_array);
-        // Only multiple tags are present
-        if (!rep_config_object_tags_map.empty() &&
-            !new_object_tags_map.empty() && prefix.empty()) {
-          if (rep_config_object_tags_map == new_object_tags_map) {
-
-            s3_log(S3_LOG_DEBUG, request_id,
-                   "Multiple tags are present - Matched with object tags\n");
-            return true;
-          }
+  }
+  for (auto dest : destination_bucket_map) {
+    std::map<std::string, std::string> dest_map = dest.second;
+
+    s3_log(S3_LOG_INFO, stripped_request_id, " %s ::[%s, %s ]\n",
+           dest.first.c_str(),
+           dest_map["DeleteMarkerReplicationStatus"].c_str(),
+           dest_map["Priority"].c_str());
+  }
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Exit\n", __func__);
+  return true;
+}
+
+bool S3ObjectMetadata::checkForFilterMatch(
+    Json::Value& rule_object,
+    std::map<std::string, std::string>& rep_config_object_tags_map,
+    const std::map<std::string, std::string>& new_object_tags_map) {
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  std::string prefix;
+  std::string key_str, val_str;
+  if (!rule_object["Filter"].isNull()) {
```

Comment:
The whole condition of this checking is toooooo large.

Please reduce the indentation.

```
if (rule_object["Filter"].isNull()) {
    s3_log(S3_LOG_DEBUG, request_id, "Empty Filter is present.\n");
    // Empty filter is specified - This means every object will be replicated.
    return true;
  }

write more code here;
```

Reply:
rule_object is json object and (rule_object["Filter"].isNull()) has two meanings -
1] "Filter" field in json file is entirely skipped
2] "Filter" field in json file is present but empty

There is no different way to check these two cases in this json parser. That is why I have to use it like this.
first if condition means "Filter" field is present and the else condition will satisfy that it does not fit in any other condition so that means its empty.